tests_e2e/tests/scripts/ext_cgroups-check_cgroups_extensions.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env pypy3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import os
+import sys
+
+from tests_e2e.tests.lib.agent_log import AgentLog
+from tests_e2e.tests.lib.cgroup_helpers import exit_if_cgroups_not_supported, print_processes, \
+    verify_agent_cgroup_assigned_correctly, BASE_CGROUP, EXT_CONTROLLERS, get_unit_cgroup_mount_path, \
+    GATESTEXT_SERVICE, AZUREMONITORAGENT_SERVICE, MDSD_SERVICE, check_quota_disabled, \
+    check_cgroup_disabled_with_unknown_process, CGROUP_TRACKED_PATTERN, AZUREMONITOREXT_FULL_NAME, GATESTEXT_FULL_NAME
+
+
+def verify_custom_script_cgroup_mount():
+    logging.info("===== Verifying custom script was assigned to the correct cgroups =====")
+
+    cpu_mounted = False
+
+    logging.info("\tcustom script cgroup mounts:")
+
+    with open('/var/lib/waagent/tmp/custom_script_check') as fh:
+        controllers = fh.read()
+        logging.info("%s", controllers)
+
+        extension_path = "/azure.slice/azure-vmextensions.slice/azure-vmextensions-Microsoft.Azure.Extensions.CustomScript"
+
+        correct_cpu_mount_v1 = "cpu,cpuacct:{0}".format(extension_path)
+        correct_cpu_mount_v2 = "cpuacct,cpu:{0}".format(extension_path)
+
+        for mounted_controller in controllers.split("\n"):
+            if correct_cpu_mount_v1 in mounted_controller or correct_cpu_mount_v2 in mounted_controller:
+                logging.info('\tCustom script extension mounted under correct cgroup '
+                      'for CPU: %s', mounted_controller)
+                cpu_mounted = True
+
+        if not cpu_mounted:
+            raise Exception('Custom script not mounted correctly! Expected {0} or {1}'.format(correct_cpu_mount_v1, correct_cpu_mount_v2))
+
+    logging.info("")
+
+
+def verify_agent_cgroup_assigned_correctly_after_ama():
+    logging.info("===== Verifying the agent cgroup consists only of agent processes after installing AMA =====")
+    with open('/var/lib/waagent/tmp/ps_check_after_ama') as fh:
+        lines = fh.readlines()
+        for process in lines:
+            logging.info(process)
+        logging.info("\n")
+
+    verify_agent_cgroup_assigned_correctly()
+
+
+def check_temporary_folder():
```

Comment:
```suggestion
def check_temporary_folder_exists():
```
nit: maybe update this name since we're only checking that it exists, not checking contents